### DISCUSSION TITLE
Unclear documentation

### DISCUSSION THREAD

**Discussion by mil-ad** [python-poetry/poetry#7115] [Q&A] [answered]
What is the difference between `poetry update --lock` and `poetry lock --no-update`? One seems to take MUCH longer for me to finish but I can't tell from the docs if they're doing anything different.

**Comment by radoering** [comment, marked as answer]
`poetry lock --no-update` tries to find a solution for the dependencies defined in pyproject.toml updating the lock file only if necessary. If nothing has been changed in pyproject.toml the lock file will not change.

`poetry update --lock` is more similar to `poetry lock` and will try to update all dependencies according to the constraints defined in pyproject.toml. That will take longer because poetry has to check if there are new versions available.

Simple example: In pyproject.toml you have a dependency like `a >= "1.0"`, version 1.0 is in the lock file, but version 1.1 is available. `poetry lock --no-update` will not change anything, `poetry update --lock` (as well as `poetry lock`) will lock 1.1. However, if you change the constraint in pyproject.toml to `a >= "1.1"`, `poetry lock --no-update` will update the lock file, too.

**Reply by neersighted** [reply]
Also see #3248

**Reply by mil-ad** [reply]
Thanks for the pointer @neersighted! I think I share the concerns raised there, it wasn't clear to me from reading the docs what the endorsed poetry workflow is when I (1) add a new dep to the toml, (2) when I switch to different git branch that has potentially different toml/lock file.

**Reply by neersighted** [reply]
If you switch to a different branch, `poetry install --sync` will exactly mirror your environment to the lock file. And if you intend to work on branches with substantial differences, [worktrees](https://git-scm.com/docs/git-worktree) are a wonderful workflow enhancement.

**Comment by mil-ad** [comment]
Thanks that makes sense. I got confused because sometimes I add a package to my toml file, followed by `poetry lock --no-update`, and see the lock file has updated completely irrelevant packages.
